### ISSUE TITLE
mixnet_delay in sim configuration

### ISSUE THREAD

**Issue by al8n** [logos-co/nomos-node#276]
> In case of using simulation, you are right about the network interface. This could be achieved by introducing new `MixnetInterface`, which uses `NetworkMessage` with a payload of `MixnetPacket`.
> 
> `MixnetPacket` could hold the information about the 3 random destinations and the actual `CarnotMessage`. If the interface receives the `MixnetPacket` and the max forward count is not exceeded, then it increases it and sends the message back to the network.

Are we sure we want to introduce this simulation wise? 
If I think straight about this, in reality adding a mixnet will just add an average delay of `X`ms to delivery. So I am not sure we actually really want to simulate this. Simplest solution would be to add a general `mixnet_delay` attribute in settings and add that to general message dispatching delay.

TLDR: Why should we actually implement mixnet delivery in the simulation app?

_Originally posted by @danielSanchezQ in https://github.com/logos-co/nomos-node/issues/273#issuecomment-1657939362_
